controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java

```diff
@@ -889,7 +916,14 @@ private TxnStatus handleDataNotFoundException(Throwable ex) {
                     final HistoryRecord newRecord = HistoryRecord.readLatestRecord(updatedTable, false).get();
                     return addIndexRecord(newRecord)
                             .thenCompose(x -> updateHistoryTable(updated))
-                            .thenCompose(x -> FutureHelpers.toVoid(updateState(State.ACTIVE)));
+                            .thenCompose(x -> FutureHelpers.toVoid(updateState(State.ACTIVE)))
+                            .whenComplete((r, e) -> {
+                                if (e != null) {
+                                    log.warn("{}/{} attempt to complete scale for epoch {}. {}", scope, name, activeEpoch, e.getClass().getName());
```

Comment:
org.slf4j.Logger#warn(java.lang.String, java.lang.Throwable) we could use the following api